src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -271,30 +276,7 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
         }
         else
         {
-            Contract.ThrowIfTrue(assetHint.DocumentId != null);
-
-            // Before doing a depth-first-search *into* each project, first run across all the project at their top
-            // level. This ensures that when we are trying to sync the projects referenced by a SolutionStateChecksums'
-            // instance that we don't unnecessarily walk all documents looking just for those.
```

Comment:
no need for this optimization. this is only for test code, so it's ok to just do a full linear walk through all proejcts.